src/js/legacy/mega-menu.js

```diff
@@ -1,23 +1,32 @@
-function isWideScreen() {
-  return matchMedia('(min-width: 768px)').matches;
-}
+import { isWideScreen, isEscape, isReverseTab, isTab } from '../common/utils/accessible-menus';
 
 class MegaMenu {
-  constructor(menuElement, openMenuElement, closeMenuElement) {
+  constructor(menuElement, openMenuElement, closeMenuElement, menuElements) {
     this.menu = menuElement;
     this.closeControl = closeMenuElement;
     this.openControl = openMenuElement;
+    this.menuElements = menuElements;
+    this.firstMenuElement = this.menuElements[0].getElementsByTagName('a')[0];
+    this.lastMenuElement = this.menuElements[0].parentNode.lastElementChild.getElementsByTagName('a')[0];
+    this.lastTabbableElement = document.querySelector('[href="http://usa.gov"]'); 
     this.addListeners = this.addListeners.bind(this);
     this.resetMenu = this.resetMenu.bind(this);
     this.showMegaMenu = this.showMegaMenu.bind(this);
     this.hideMegaMenu = this.hideMegaMenu.bind(this);
+    this.enterSmallMegaMenu = this.enterSmallMegaMenu.bind(this);
+    this.exitSmallMegaMenu = this.exitSmallMegaMenu.bind(this);
+    this.toggleSmallMegaMenu = this.toggleSmallMegaMenu.bind(this);
 
     this.addListeners();
   }
 
   addListeners() {
     this.closeControl.addEventListener('click', this.hideMegaMenu);
+    this.closeControl.addEventListener('keydown', this.enterSmallMegaMenu);
     this.openControl.addEventListener('click', this.showMegaMenu);
+    this.firstMenuElement.addEventListener('keydown', this.exitSmallMegaMenu);
+    this.lastMenuElement.addEventListener('keydown', this.exitSmallMegaMenu);
+    this.menuElements.forEach(item => item.addEventListener('keydown', this.toggleSmallMegaMenu));
```

Comment:
What do you think about having one event listener on `#vetnav-menu` and checking that the target is an `li`?

Reply:
That's definitely a better idea, thanks @jbalboni, changed.